src/acl.rs

```diff
@@ -0,0 +1,269 @@
+//! Defines types to use with the ACL commands.
+
+use crate::types::{ErrorKind, FromRedisValue, RedisResult, RedisWrite, ToRedisArgs, Value};
+
+macro_rules! not_convertible_type_error {
+    ($v:expr, $det:expr) => {{
+        fail!((
+            ErrorKind::TypeError,
+            "Response type not convertible to Rule",
+            format!("{:?} (response was {:?})", $det, $v)
+        ))
+    }};
+}
+
+/// ACL rules are used in order to activate or remove a flag, or to perform a
+/// given change to the user ACL, which under the hood are just single words.
+#[derive(Debug, Eq, PartialEq)]
+pub enum Rule {
+    /// Enable the user: it is possible to authenticate as this user.
+    On,
+    /// Disable the user: it's no longer possible to authenticate with this
+    /// user, however the already authenticated connections will still work.
+    Off,
+
+    /// Add the command to the list of commands the user can call.
+    AddCommand(String),
+    /// Remove the command to the list of commands the user can call.
+    RemoveCommand(String),
+    /// Add all the commands in such category to be called by the user.
+    AddCategory(String),
+    /// Remove the commands from such category the client can call.
+    RemoveCategory(String),
+    /// Alias for `+@all`. Note that it implies the ability to execute all the
+    /// future commands loaded via the modules system.
+    AllCommands,
+    /// Alias for `-@all`.
+    NoCommands,
+
+    /// Add this password to the list of valid password for the user.
+    AddPass(String),
+    /// Remove this password from the list of valid passwords.
+    RemovePass(String),
+    /// Add this SHA-256 hash value to the list of valid passwords for the user.
+    AddHashedPass(String),
+    /// Remove this hash value from from the list of valid passwords
+    RemoveHashedPass(String),
+    /// All the set passwords of the user are removed, and the user is flagged
+    /// as requiring no password: it means that every password will work
+    /// against this user.
+    NoPass,
+    /// Flush the list of allowed passwords. Moreover removes the _nopass_ status.
+    ResetPass,
+
+    /// Add a pattern of keys that can be mentioned as part of commands.
+    Pattern(String),
+    /// Alias for `~*`.
+    AllKeys,
+    /// Flush the list of allowed keys patterns.
+    ResetKeys,
+
+    /// Performs the following actions: `resetpass`, `resetkeys`, `off`, `-@all`.
+    /// The user returns to the same state it has immediately after its creation.
+    Reset,
+}
+
+impl ToRedisArgs for Rule {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        use self::Rule::*;
+
+        let rule = match self {
+            On => "on".to_owned(),
+            Off => "off".to_owned(),
+
+            AddCommand(cmd) => "+".to_owned() + cmd,
```

Comment:
Created https://github.com/mitsuhiko/redis-rs/pull/348 which would allow you to avoid these allocations

```
out.write_arg_fmt(format_args!("+{}", cmd))
```